statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,257 @@
+import numpy as np 
+import pandas as pd
+#from numbers import Number
+
+
+class SurveyStat(object):
+
+    def __init__(self, data, cluster=None, strata=None, prob_weights=None):
+
+        data, self.cluster = self._check_type(data, cluster)
+        data, self.strata = self._check_type(data, strata)
+        data, self.prob_weights = self._check_type(data, prob_weights)
+        
+        self.p = data.shape[1]
+        self.data = np.asarray(data)
+
+    # preprocessing to be used by init
+    def _check_type(self, data, vname):
+        """
+        converts cluster, strata, etc to an numpy array
+        gets rid of column from data if possible
+
+        Parameters
+        ----------
+        data : array-like
+          The raw data
+        vname : one of cluster, strata, or prob_weight
+
+        Returns
+        -------
+        A nx1 numpy array
+        """
+        # check if str, num, or list
+
+        self.n = data.shape[0]
+        if isinstance(vname, str):
+            temp = data[vname].values
+            del data[vname]
+            vname = temp
+            n = len(vname)
+            vname = vname.reshape(self.n,1)  
+        elif isinstance(vname, list):
```

Comment:
I think the 2nd and 4th clauses in this if block can be combined, just using `np.asarray(vname)`.  I don't think the `reshape` is needed here.  `np.asarray` is more efficient than `np.array` and can be safely called on a list, ndarray, or pandas object.